catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/activity/MainMenuActivityButtonsTest.java

```diff
@@ -0,0 +1,131 @@
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.uiespresso.ui.activity;
+
+import android.support.test.espresso.IdlingResource;
+import android.support.test.runner.AndroidJUnit4;
+
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.ui.MainMenuActivity;
+import org.catrobat.catroid.ui.WebViewActivity;
+import org.catrobat.catroid.uiespresso.util.BaseActivityInstrumentationRule;
+import org.catrobat.catroid.uiespresso.util.UiTestUtils;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.lang.reflect.InvocationTargetException;
+
+import static android.support.test.espresso.Espresso.registerIdlingResources;
+import static android.support.test.espresso.Espresso.unregisterIdlingResources;
+import static android.support.test.espresso.Espresso.pressBack;
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+import static junit.framework.Assert.assertTrue;
+
+@RunWith(AndroidJUnit4.class)
+public class MainMenuActivityButtonsTest {
+	private IdlingResource mainMenuIdlingResource;
+
+	@Rule
+	public BaseActivityInstrumentationRule<MainMenuActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(MainMenuActivity.class, true, false);
+
+	@Before
+	public void setUp() throws Exception {
+		baseActivityTestRule.launchActivity(null);
+		mainMenuIdlingResource = baseActivityTestRule.getActivity().getIdlingResource();
+		registerIdlingResources(mainMenuIdlingResource);
+	}
+
+	@Test
+	public void buttonContinueOpensProjectActivity() throws ClassNotFoundException,
+			NoSuchMethodException, NoSuchFieldException,
+			InvocationTargetException, IllegalAccessException {
+		onView(withId(R.id.main_menu_button_continue))
+				.perform(click());
+
+		onView(withId(R.id.fragment_sprites_list))
+				.check(matches(isDisplayed()));
+	}
+
+	@Test
+	public void buttonNewOpensDialog() {
+		onView(withId(R.id.main_menu_button_new))
+				.perform(click());
+
+		onView(withText(R.string.new_project_dialog_title))
+				.check(matches(isDisplayed()));
+	}
+
+	@Test
+	public void buttonProgramsOpensProgramsList() {
+		onView(withId(R.id.main_menu_button_programs))
+				.perform(click());
+
+		onView(withText(R.string.programs))
+				.check(matches(isDisplayed()));
+	}
+
+	@Test
+	public void buttonHelpWithConnectionCallsOpenHelpSite() {
+		onView(withId(R.id.main_menu_button_help))
+				.perform(click());
+
+		assertTrue(UiTestUtils.getCurrentActivity() instanceof WebViewActivity);
```

Comment:
it would be even nicer if it is checked that the right url is loaded. this could be done on two different ends, either verifying that the intend thats opening webViewActivity has the right payload (https://google.github.io/android-testing-support-library/docs/espresso/intents/) or on the other end, adding a getUrl function to WebViewactivity and polling this and asserting.